src/Elmish.WPF/Binding.fs

```diff
@@ -1684,6 +2148,35 @@ type Binding private () =
       canExec
       (defaultArg uiBoundCmdParam false)
 
+  /// <summary>
+  ///   Creates a <c>Command</c> binding that depends on the
+  ///   <c>CommandParameter</c>
+  ///   and can execute if <paramref name="canExec" /> returns <c>true</c>.
+  /// </summary>
+  /// <param name="exec">Returns the message to dispatch.</param>
+  /// <param name="canExec">Indicates whether the command can execute.</param>
+  /// <param name="uiBoundCmdParam">
+  ///   If <c>true</c>, <c>CanExecuteChanged</c> will trigger every time WPF's
+  ///   <c>CommandManager</c>
+  ///   detects UI changes that could potentially influence the command's
+  ///   ability to execute. This will likely lead to many more triggers than
+  ///   necessary, but is needed if you have bound the <c>CommandParameter</c>
+  ///   to another UI property.
+  /// </param>
+  /// <param name="wrapDispatch">
+  ///   Wraps the dispatch function with additional behavior, such as
+  ///   throttling, debouncing, or limiting.
+  /// </param>
+  [<System.Obsolete("In version 5, this method will be removed.  Use the overload without the \"wrapDispatch\" parameter followed by a call to \"Binding.alterMsgStream\".  For an example, see how this method is implemented.")>]
+  static member cmdParamIf
+      (exec: obj -> 'model -> 'msg,
+       canExec: obj -> 'model -> bool,
+       ?uiBoundCmdParam: bool,
+       ?wrapDispatch: Dispatch<'msg> -> Dispatch<'msg>)
+      : string -> Binding<'model, 'msg> =
+    Binding.cmdParamIf (exec, canExec, defaultArg uiBoundCmdParam false)
+    >> Binding.alterMsgStream (defaultArg wrapDispatch id)
```

Comment:
This is causing `alterMsgStream` to be called everytime (sometimes with `id`), it might be better to check if the arg is defined and only call the alter then.

Reply:
It could be something like

```suggestion
    >> match wrapDispatch with | Some wrapDispatch -> Binding.alterMsgStream wrapDispatch | None -> id
```

Or we could add some helper like `mapBoth` or whatever name you prefer.

Reply:
It can be made more terse:
```fsharp
>> (wrapDispatch |> Option.map Binding.alterMsgStream |> Option.defaultValue id)
```